src/vanilla/DanModel/DanCodeGenerator.cs

```diff
@@ -6445,5 +6477,250 @@ internal static string IModelTypeDefaultValue(IModelType modelType, Method paren
+            {
+                result = parameter.Name;
+            }
+            else
+            {
+                string caller = (parameter.Method != null && true == parameter.Method.Group.IsNullOrEmpty() ? "this" : "this.client");
+                result = $"{caller}.{parameter.ClientProperty.Name.ToCamelCase()}()";
+            }
+            return result;
+        }
+
+        internal static void ParameterSetName(Parameter parameter, string name)
+        {
+            parameter.Name = name;
+        }
+
+        internal static IModelType ParameterGetModelType(Parameter parameter)
+        {
+            IModelType result = parameter.ModelType;
+            if (result != null && !ParameterWantNullable(parameter))
+            {
+                result = GetIModelTypeNonNullableVariant(result);
+            }
+            return result;
+        }
+
+        internal static void ParameterSetModelType(Parameter parameter, IModelType modelType)
+        {
+            parameter.ModelType = modelType;
+        }
+
+        internal static IModelType ParameterClientType(Parameter parameter)
+            => GetIModelTypeParameterVariant(ParameterGetModelType(parameter));
+
+        internal static IModelType ParameterWireType(Parameter parameter)
+        {
+            IModelType modelType = ParameterGetModelType(parameter);
+            if (modelType.IsPrimaryType(KnownPrimaryType.Stream))
+            {
+                // Just pass whatever we give to clients to RestProxy
+                return ParameterClientType(parameter);
+            }
+            else if (!modelType.IsPrimaryType(KnownPrimaryType.Base64Url) &&
+                parameter.Location != ParameterLocation.Body &&
+                parameter.Location != ParameterLocation.FormData &&
+                NeedsSpecialSerialization(ParameterClientType(parameter)))
+            {
+                return DependencyInjection.New<PrimaryType>(KnownPrimaryType.String);
+            }
+            else
+            {
+                return modelType;
+            }
+        }
+
+        internal static bool ParameterNeedsConversion(Parameter parameter)
+            => !ParameterClientType(parameter).StructurallyEquals(ParameterWireType(parameter));
+
+        internal static string ParameterWireName(Parameter parameter)
+            => ParameterNeedsConversion(parameter) ? $"{ParameterGetName(parameter).ToCamelCase()}Converted" : ParameterGetName(parameter);
+
+        private static string ParameterConvertToWireType(Parameter parameter, string source, string clientReference)
+        {
+            if (parameter.Location != ParameterLocation.Body &&
+                parameter.Location != ParameterLocation.FormData &&
+                NeedsSpecialSerialization(ParameterGetModelType(parameter)))
+            {
+                IModelType clientType = ParameterClientType(parameter);
+                PrimaryType primary = clientType as PrimaryType;
```

Comment:
No rush but consider changing these to `is PrimaryType primary` when you see them